src/d3a/models/strategy/external_strategies/pv.py

```diff
@@ -212,22 +221,25 @@ def _area_reconfigure_prices(self, **kwargs):
         if not self.connected:
             super()._area_reconfigure_prices(**kwargs)
 
+    def _incoming_commands_callback_selection(self, req):
+        if req.request_type == "offer":
+            self._offer_impl(req.arguments, req.response_channel)
+        elif req.request_type == "delete_offer":
+            self._delete_offer_impl(req.arguments, req.response_channel)
+        elif req.request_type == "list_offers":
+            self._list_offers_impl(req.arguments, req.response_channel)
+        elif req.request_type == "device_info":
+            self._device_info_impl(req.arguments, req.response_channel)
+        else:
+            assert False, f"Incorrect incoming request name: {req}"
+
     def event_tick(self):
         if not self.connected and not self.is_aggregator_controlled:
             super().event_tick()
         else:
             while len(self.pending_requests) > 0:
                 req = self.pending_requests.pop()
-                if req.request_type == "offer":
-                    self._offer_impl(req.arguments, req.response_channel)
-                elif req.request_type == "delete_offer":
-                    self._delete_offer_impl(req.arguments, req.response_channel)
-                elif req.request_type == "list_offers":
-                    self._list_offers_impl(req.arguments, req.response_channel)
-                elif req.request_type == "device_info":
-                    self._device_info_impl(req.arguments, req.response_channel)
-                else:
-                    assert False, f"Incorrect incoming request name: {req}"
+                self._incoming_commands_callback_selection(req)
```

Comment:
Good to be abstracted into a method but can you help me understand the need to be called from `event_activate ` too?

Reply:
I havent added it to the `event_activate`. Or can you point me to it please?

Reply:
Sorry yes you are right, my bad.